### ISSUE TITLE
File names with brackets are ignored

### ISSUE THREAD

**Issue by jowsey** [standard/vscode-standard#187]
**What version of this package are you using?**
v1.5.0

**What operating system, Node.js, and npm version?**
Windows 10
v15.11.0 
7.6.0

**What happened?**
When working with files that include special characters like brackets, the extension explicitly ignores them, for some reason
For example, `epicTextFile.js` would be registered, but `[epicTextFile].js` would not.

**What did you expect to happen?**
The extension should treat all files the same and lint them regardless of special characters

**Are you willing to submit a pull request to fix this bug?**
I don't understand TypeScript / this codebase enough to be very helpful but sure

These are my extension settings
```json
  // StandardJS
  "javascript.validate.enable": false,
  "typescript.validate.enable": false,
  "standard.enable": true,
  "standard.usePackageJson": true,
  "standard.autoFixOnSave": true,
  "standard.validate": [
    "javascript",
    "javascriptreact",
    "typescript",
    "typescriptreact"
  ],
```

**Comment by theoludwig** [comment]
Thanks for your report! @jowsey 

The issue has already been fixed, I just tested to be sure, and for me it's working perfectly.
![image](https://user-images.githubusercontent.com/25207499/112422981-124f4c00-8d32-11eb-9555-db9d3442a042.png)

Could you make a minimal reproductible repo, please ?

**Comment by jowsey** [comment]
> Could you make a minimal reproductible repo, please ?

This happens in every file with brackets, regardless of the repo or environment. The example image you sent doesn't work for me. 
Another thing that's interesting is that the file reported in the terminal is not the same as the actual file name.

I get this every time I save the file `c:\Users\Jowsey\Desktop\random stuff\[epicTextFile[]].js ignored.`
Notice the [] at the end, that is not in the file name

I'm definitely using version 1.5.0 so this is quite confusing

**Comment by theoludwig** [comment]
Right, sorry I can't reproduce so I can't help there, I'm also using the `1.5.0` version (latest).
We'll reopen this issue, if many people report the same problem, it is a duplicate of #102.

**Comment by lukas-sl** [comment]
I stumbled across the same issue when using the extension while working on a Next.js project.

Here are two screenshots displaying the error:
![image](https://user-images.githubusercontent.com/5834134/128164157-2c82b59e-849e-445e-b734-badb4943e489.png)
![image](https://user-images.githubusercontent.com/5834134/128164245-1efbaa0f-d51e-433c-a7d9-5774f1dd4636.png)


What version of this package are you using?
v1.5.1

What operating system, Node.js, and yarn version?
Windows 10
v15.0.1
1.22.4

**Comment by theoludwig** [comment]
Hey! :wave: @lukas-sl 

Thanks for your report!
I still can't reproduce the issue.
![image](https://user-images.githubusercontent.com/25207499/128325544-b0a87104-0fff-41d5-8e6b-ad587aee561d.png)

**Comment by yrokhlin** [comment]
This still persists to this day.
Specifically a next.js project, will always fail if the file has brackets, double brackets, or the folder name has brackets.

**Comment by josegutierro** [comment]
Hello, 

In my case the problem exists when a bracket file is inside a bracket folder, for example:

`/pages/[id].js` -> It works
`/pages/blog/[id].js` -> Also it works
`/pages/[blogId]/[id].js` -> It does not work

**Comment by josegutierro** [comment]
any update on this?

**Comment by theoludwig** [comment]
Excuse me for the late reply @josegutierro!
Thanks for your investigation, indeed I just tried to reproduce the issue again and your examples are correct. :+1:
It doesn't work when a bracket file is inside a bracket folder, but if it's only the folder or only the filename with brackets it works.

I reopened the issue as this is a confirmed bug, sorry if we closed it even if it wasn't completely fixed, we closed it because we didn't know how to reproduce the issue.

Concerning the fix, I will try when I have some time to fix it, but we always accept PRs and welcome new contributors. :smile:
Maybe this commit can help to fix the issue: https://github.com/standard/vscode-standard/commit/6f17ebb609f41faca0d30d2190672f1c4f7915d7

**Comment by josegutierro** [comment]
Thank you so much @Divlo. Maybe this will help you, I think is also related with the eslint issue: https://github.com/microsoft/vscode-eslint/issues/977

Anything that I can help you with just let me know!

**Comment by josegutierro** [comment]
Btw! You said that commit https://github.com/standard/vscode-standard/commit/6f17ebb609f41faca0d30d2190672f1c4f7915d7 can help me with the issue. How could I install this version on my VSCode to test?

Thank you!

**Comment by theoludwig** [comment]
@josegutierro You can temporarily uninstall the published extension version and follow the instructions here: https://github.com/standard/vscode-standard#how-to-develop.

**Comment by josegutierro** [comment]
Hello @Divlo the issue still remains... with commit https://github.com/standard/vscode-standard/commit/6f17ebb609f41faca0d30d2190672f1c4f7915d7

**Comment by josegutierro** [comment]
Hello @Divlo I create the PR that seems to solve the issue https://github.com/standard/vscode-standard/pull/443

**Comment by theoludwig** [comment]
> Hello @Divlo I create the PR that seems to solve the issue #443

Thanks, it works! :smile:
It has been merged, but unfortunately, there is an issue to publish new versions on the VSCode marketplace (see: <https://github.com/standard/vscode-standard/issues/444>), as soon as it has been resolved, we will publish v2.1.1. :+1:

**Comment by josegutierro** [comment]
Hello @Divlo happy to hear that!

In the meanwhile I built it my own .vsix and seems to work fine. I will wait the official VSCode marketplace publish to install the version 2.1.1 😃
